src/main/java/pl/lodz/zzpj/kanbanboard/service/TaskService.java

```diff
@@ -0,0 +1,188 @@
+package pl.lodz.zzpj.kanbanboard.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import pl.lodz.zzpj.kanbanboard.entity.Review;
+import pl.lodz.zzpj.kanbanboard.entity.Task;
+import pl.lodz.zzpj.kanbanboard.entity.Task.Status;
+import pl.lodz.zzpj.kanbanboard.entity.TaskDetails;
+import pl.lodz.zzpj.kanbanboard.entity.TaskDetails.Difficulty;
+import pl.lodz.zzpj.kanbanboard.entity.User;
+import pl.lodz.zzpj.kanbanboard.exceptions.BadOperationException;
+import pl.lodz.zzpj.kanbanboard.exceptions.ConflictException;
+import pl.lodz.zzpj.kanbanboard.exceptions.NotFoundException;
+import pl.lodz.zzpj.kanbanboard.repository.ReviewsRepository;
+import pl.lodz.zzpj.kanbanboard.repository.TaskDetailsRepository;
+import pl.lodz.zzpj.kanbanboard.repository.TasksRepository;
+import pl.lodz.zzpj.kanbanboard.repository.UsersRepository;
+import pl.lodz.zzpj.kanbanboard.utils.DateProvider;
+
+import java.time.Instant;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+public class TaskService extends BaseService {
+
+    private final TasksRepository tasksRepository;
+
+    private final TaskDetailsRepository taskDetailsRepository;
+
+    private final ReviewsRepository reviewsRepository;
+
+    private final UsersRepository usersRepository;
+
+    private final DateProvider dateProvider;
+
+    @Autowired
+    public TaskService(
+            TasksRepository tasksRepository, TaskDetailsRepository taskDetailsRepository,
+            ReviewsRepository reviewsRepository, UsersRepository usersRepository, DateProvider dateProvider) {
+        this.tasksRepository = tasksRepository;
+        this.taskDetailsRepository = taskDetailsRepository;
+        this.reviewsRepository = reviewsRepository;
+        this.usersRepository = usersRepository;
+        this.dateProvider = dateProvider;
+    }
+
+    public Optional<Task> getTaskByUUID(UUID taskUuid) {
+        return tasksRepository.findByUuid(taskUuid);
+    }
+
+    public List<Task> getAll() {
+        return tasksRepository.findAll();
+    }
+
+    public List<Task> getAllTasksCreatedBy(String userEmail) {
+        return getAll()
```

Comment:
Filtering data should be done in the repository. Spring data automatically implements those.